libs/libevhtp/Makefile

```diff
@@ -25,11 +25,7 @@ PKG_MAINTAINER:=Eneas U de Queiroz <cote2004-github@yahoo.com>
 PKG_LICENSE:=BSD-3-Clause
 PKG_LICENSE_FILES:=LICENSE
 
-TARGET_CFLAGS += $(TARGET_CPPFLAGS)
 CMAKE_INSTALL:=1
-CMAKE_OPTIONS += \
-	-DLIB_INSTALL_DIR="$(STAGING_DIR)/usr/lib" \
-	-DINCLUDE_INSTALL_DIR="$(STAGING_DIR)/usr/include"
```

Comment:
Contents:

```
prefix=/usr
libdir=/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib
includedir=/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/include/evhtp

Name: libevhtp
Description: A more flexible replacement for libevent's httpd API
Version: 1.2.18
Libs: -L${libdir} -levhtp
Libs.private: /home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent.so;/home$Cflags: -I${includedir}
```

looks correct.

Reply:
I figured out what happened.  When these are defined, `make clean` will remove them, so we need to leave them out, but then edit `evhtp.pc` so that we don't end up with an empty `-L`.
```Makefile
define Build/InstallDev
	$(call Build/InstallDev/cmake,$(1))
	$(SED) 's,^\(include\|lib\)dir=,\0$(STAGING_DIR)/usr/\1,' "$(1)/usr/lib/pkgconfig/evhtp.pc"
endef
```

Reply:
Updated.

Reply:
Mine turns out like this without the code above:
```
prefix=/usr
libdir=
includedir=/evhtp

Name: libevhtp
Description: A more flexible replacement for libevent's httpd API
Version: 1.2.18
Libs: -L${libdir} -levhtp
Libs.private: /home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libevent.so;/home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libevent_openssl.so;/home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libevent_core.so;/home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libevent_extra.so;/home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libevent_pthreads.so;/home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libevent_extra.so;OpenSSL::SSL;OpenSSL::Crypto;Threads::Threads;/home/equeiroz/src/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_musl_eabi/usr/lib/libonig.so
Cflags: -I${includedir}
```

Reply:
⚠️  We need to figure out why we end up with different `evhtp.pc` files, first and foremost.

Reply:
Looks like my file is a holdover from a prior version. I will retest.

Reply:
Yeah. With the original commit here I get:

```
prefix=/usr
libdir=/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib
includedir=/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/include/evhtp

Name: libevhtp
Description: A more flexible replacement for libevent's httpd API
Version: 1.2.18
Libs: -L${libdir} -levhtp
Libs.private: /home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent.so;/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent_openssl.so;/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent_core.so;/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent_extra.so;/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent_pthreads.so;/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libevent_extra.so;OpenSSL::SSL;OpenSSL::Crypto;Threads::Threads;/home/mangix/devstuff/openwrt/staging_dir/target-arm_cortex-a9+vfpv3_glibc_eabi/usr/lib/libonig.so
Cflags: -I${includedir}
```

Reply:
😌 
Good to hear you got a bad file. 🤣

Reply:
And with the SED fix I get:

```
prefix=/usr
libdir=/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib
includedir=/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/include/evhtp

Name: libevhtp
Description: A more flexible replacement for libevent's httpd API
Version: 1.2.18
Libs: -L${libdir} -levhtp
Libs.private: /home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libevent.so;/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libevent_openssl.so;/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libevent_core.so;/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libevent_extra.so;/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libevent_pthreads.so;/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libevent_extra.so;OpenSSL::SSL;OpenSSL::Crypto;Threads::Threads;/home/mangix/devstuff/openwrt/staging_dir/target-arc_arc700_uClibc/usr/lib/libonig.so
Cflags: -I${includedir}
```

Reply:
This should work.  It is exactly what you had before, isn't it?  I don't think `prefix` does anything here except making it look bad.

Reply:
The private libs are more numerous from what I see. Talking about the very first file. In most other files in staging_dir/usr/lib/pkgconfig , prefix is set to /usr. Should not be a problem.